### DISCUSSION TITLE
Client event not received from nodejs socketio server

### DISCUSSION THREAD

**Discussion by pluzmedia** [miguelgrinberg/python-socketio#803] [Q&A]
I can successfully emit to NodeJS socketio from python client by the following code
```sio.emit("coords", {"id": rowId})```

but emits from the NodeJS sever are not received by python client

NodeJS server code 
```socket.emit("catch_all", res.id)```

Python Client code
```
@sio.on('catch_all')
def catch_all(data):
    print('message received with ', data)
    sio.emit('coords', {'response': 'my response'})
```

or
 ```
@sio.event
def catch_all(data):
    print('message received with ', data)
    sio.emit('coords', {'response': 'my response'})
```
Any help?

**Comment by miguelgrinberg** [comment]
First of all, what is the logic in writing a bug report against this package when you really just have a question and need help? Bug reports are that, bug reports. If you don't have a bug to report, then don't write one. There is a discussion board that is specifically there for questions like yours.

Now, with regards to your question, I need to see your logs. The troubleshooting section of the documentation explains how to enable logs. Please do that, then add the logs to your question.

**Reply by pluzmedia** [reply]
Thanks for the proper direction. I have pasted the logs below

```
Attempting polling connection to http://domain.ttl:4000/socket.io/?transport=polling&EIO=4
<class 'socketio.exceptions.BadNamespaceError'>
Polling connection accepted with {'sid': 'Uhh5CZyOp_HO0FxbAAAC', 'upgrades': ['websocket'], 'pingInterval': 25000, 'pingTimeout': 20000}
Engine.IO connection established
Sending packet MESSAGE data 0
Attempting WebSocket upgrade to ws://domain.ttl:4000/socket.io/?transport=websocket&EIO=4
<class 'socketio.exceptions.BadNamespaceError'>
WebSocket upgrade was successful
Received packet MESSAGE data 0{"sid":"VkveaiUpelyEbQ-oAAAD"}
Namespace / is connected
connection established
Emitting event "coords" [/]
Sending packet MESSAGE data 21["coords",{"id":318,"json":{"lat":0.0,"lng":0.0,"speed":null}}]
Emitting event "coords" [/]
Sending packet MESSAGE data 22["coords",{"id":319,"json":{"lat":0.0,"lng":0.0,"speed":null}}]
Emitting event "coords" [/]
Sending packet MESSAGE data 23["coords",{"id":320,"json":{"lat":0.0,"lng":0.0,"speed":null}}]
Emitting event "coords" [/]
Sending packet MESSAGE data 24["coords",{"id":321,"json":{"lat":0.0,"lng":0.0,"speed":null}}]
Emitting event "coords" [/]
Sending packet MESSAGE data 25["coords",{"id":322,"json":{"lat":0.0,"lng":0.0,"speed":null}}]
Emitting event "coords" [/]
Sending packet MESSAGE data 26["coords",{"id":323,"json":{"lat":0.0,"lng":0.0,"speed":null}}]
Emitting event "coords" [/]
Sending packet MESSAGE data 27["coords",{"id":324,"json":{"lat":0.0,"lng":0.0,"speed":null}}]
Emitting event "coords" [/]
Sending packet MESSAGE data 28["coords",{"id":325,"json":{"lat":0.0,"lng":0.0,"speed":null}}]
```

**Reply by miguelgrinberg** [reply]
The problem is in your client. There is nothing in these logs showing that the client is sending.

**Reply by pluzmedia** [reply]
I have using latest python-socketio client. How to address this issue and insight would b more helpful. And one more thing, this client is running on a Thread.

Update: I tried running outside the Thread but no luck
Update 2: connect & disconnect events are working fine

**Reply by miguelgrinberg** [reply]
Sorry I said client, I meant server. The Node side is not sending, that's what I meant to say. Sorry for any confusion.

**Reply by pluzmedia** [reply]
NodeJS server recives events find and emit back to the socket but the python client didn't receive the events. Is there anyway to address this?. NodeJS client works fine.
```
socket.on("coords", async function (res, callback) {
        //let data = JSON.parse(res)
        socket.emit("catch_all", res.id)
        callback(res.id)
        console.log(res);
    });
```

**Reply by miguelgrinberg** [reply]
if the client doesn't log, then the server isn't sending. Even if the server is sending garbage or invalid stuff, the client should log it, and it isn't logging anything. So your next step is to debug your server, and make sure it is sending. If you have a complete example that demonstrates that this is a bug in the client, then provide the code and I'll look at it.

**Reply by pluzmedia** [reply]
NodeJS socket.io version is ```socket.io@4.2.0```
Python-socketio client version is ```5.4.1```

Are those compatible?

**Reply by miguelgrinberg** [reply]
Also, what are these lines in your log?

    <class 'socketio.exceptions.BadNamespaceError'>

Is your application printing these lines?

**Reply by pluzmedia** [reply]
yes and as you said the problem lies in the server. I have created separate namespace for this application and it is working great. Thanks for your kindness.
